### PR TITLE
Replacing theme variables

### PR DESCRIPTION
**Pull request by ludwiczakpawel**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [x] Optimization

## Description

**TL;DR Replacing old variables in themes with new ones.**

This PR is just another step towards implementing Crayons - it cleans up our variables a bit more unblocking us to customize components based on applied theme. I got rid of most of the old variables and replaced them with new ones. There're couple more left but I intentionally kept them since removing them would affect visuals too much. But we will remove them step by step while implementing Crayons components.

One thing to keep in mind is that this PR will affect visually lots of places. Changes, in most cases, should not be too drastic and they should definitely not break things. I'm pretty sure we will find UI glitches and some tiny issues but I'd rather ship this and fix everything gradually rather than split it into 341 separate tiny PRs because it would take 17 years. 

## Added tests?

- [ ] yes
- [ ] no, because they aren't needed
- [ ] no, because I need help

## Added to documentation?

- [x] no documentation needed

### CODE REVIEW

File: app/assets/stylesheets/article-form-needed-legacy.scss [hunk 1]
```diff
@@ -209,25 +209,6 @@
     }
   }
 
-  .org-publish-check-wrapper {
```
Comment (ludwiczakpawel) [inline app/assets/stylesheets/article-form-needed-legacy.scss]: It was never used.

File: app/assets/stylesheets/article-show.scss [hunk 2]
```diff
@@ -1,10 +1,6 @@
 @import 'variables';
 @import 'mixins';
 
-.stories-show {
-  @include themeable(background, theme-background, $lightest-gray);
```
Comment (ludwiczakpawel) [inline app/assets/stylesheets/article-show.scss]: I found bunch of cases like this, where styling was kinda redundant and unnecessary.

File: app/assets/stylesheets/article-show.scss [hunk 3]
```diff
@@ -779,24 +763,9 @@ article {
         font-size: 0.85em;
 
         a {
-          @include themeable(color, theme-secondary-color, $medium-gray);
+          color: var(--container-color-tertiary);
           margin-right: 6px;
           display: inline-block;
-
-          .icon-img {
```
Comment (ludwiczakpawel) [inline app/assets/stylesheets/article-show.scss]: I basically replaced `` in couple views with inline SVGs so it's easier to style them depending on theme.

**Comment by benhalpern** [comment]
This looks really great!

I didn't see any visual bugs except one I realized was already in prod from another push 🙃

I know that this is primarily an infrastructure PR, but the purple WRITE A POST button really classes up the joint!

**Comment by rhymes** [comment]
I'm unsure as to why the test fails, as it doesn't locally. I'll try to restart the build again

**Comment by ludwiczakpawel** [comment]
@rhymes hmm it doesn't fail locally for me as well soooo... 🤷‍♂️

**Comment by rhymes** [comment]
@maestromac / @mstruve given the test failure [here](https://travis-ci.com/github/thepracticaldev/dev.to/builds/154252008#L1870) and since none of us can reproduce it, is there a way to connect to the Travis box and get ahold of the HTML?

File: app/views/organizations/_sidebar_additional.html.erb [hunk 4]
```diff
@@ -2,7 +2,7 @@
   <div class="sidebar-bg" id="sidebar-bg-right"></div>
   <div class="side-bar sidebar-additional showing" id="sidebar-additional">
     <% @organization.sponsorships.live.find_each do |sponsorship| %>
-      <div class="sidebar-sponsorship-level" style="background:<%= sponsorship.decorate.level_background_color %>">
```
Comment (nickytonline) [inline app/views/organizations/_sidebar_additional.html.erb]: Check with @benhalpern about this. They probably change this in the database when changing sponsors.

Comment (benhalpern) [reply app/views/organizations/_sidebar_additional.html.erb]: Yeah, but if we lose this from the design it's fine with me.

Comment (benhalpern) [reply app/views/organizations/_sidebar_additional.html.erb]: Good call though

File: app/assets/stylesheets/config/_colors.scss [hunk 5]
```diff
@@ -51,8 +51,8 @@
   ////////////////////////////////////////////////////////////////////////////////////
 
   // Main colors
-  --body-bg: var(--base-10);
-  --body-color: var(--base-100);
+  --body-bg: #f9fafa; // todo: replace with `var(--base-10);` when ready.
```
[the remark this answers is not in the document]
Comment (ludwiczakpawel) [reply app/assets/stylesheets/config/_colors.scss]: Correct. We will ship new styling step by step and then introduce these colors :)

**Comment by mstruve** [comment]
@rhymes there is currently no way to get that HTML, we only store snapshots that are in the top folder `tmp/screenshots/*.png` and are images. Maybe run Travis in debug mode and ssh onto the box to see if you can recreate it and throw in some pry statements

**Comment by rhymes** [comment]
@mstruve thanks!

@nickytonline thank you yeah, there's a conflict right now. I think you also need an admin champion to have this merged if Travis needs to be bypassed

File: app/assets/stylesheets/article-show.scss [hunk 6]
```diff
@@ -1220,14 +1171,7 @@ article {
...
             cursor: pointer !important;
-            @include themeable-important(color, theme-color, $black);
-            &:hover {
-              @include themeable-important(
-                background,
-                theme-top-bar-background-hover,
-                $light-gray
-              );
-            }
+            color: var(--body-color);
```
[the remark this answers is not in the document]
Comment (ludwiczakpawel) [reply app/assets/stylesheets/article-show.scss]: Surprisingly, it works just fine in this case without `!important` 🤷‍♂️ But yea, I'll keep an eye on that mixin.

**Comment by ludwiczakpawel** [comment]
@rhymes @nickytonline conflict resolved, latest master merged in.

File: app/assets/stylesheets/article-show.scss (hunk 6 above)
[the remark this answers is not in the document]
Comment (nickytonline) [reply app/assets/stylesheets/article-show.scss]: Sounds good, it’s possible that other improvements to the CSS in other PRs have fixed the usage.

**Comment by rhymes** [comment]
Ok, we're back to that single test failing
